scipy/signal/_ltisys.py

```diff
@@ -3552,8 +3552,7 @@ def dlsim(system, u, t=None, x0=None):
         if len(u.shape) == 1:
             u = u[:, np.newaxis]
 
-        u_dt_interp = interp1d(t, u.transpose(), copy=False, bounds_error=True)
```

Comment:
Unless I'm missing something simple, it does not extrapolate really. This change is in the `if t is not None` branch, and just a few lines above, `tout` is defined like this

```
    if t is None:
        out_samples = len(u)
        stoptime = (out_samples - 1) * system.dt
    else:
        stoptime = t[-1]
        out_samples = int(np.floor(stoptime / system.dt)) + 1
    # Pre-build output arrays
    <snip>
    tout = np.linspace(0.0, stoptime, num=out_samples)
```

so that `tout[-1] == t[-1]` and there is no extrapolation.